firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigRealtimeHttpClient.java

```diff
@@ -0,0 +1,67 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+//
+// You may obtain a copy of the License at
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.remoteconfig.internal;
+
+import com.google.firebase.remoteconfig.ConfigUpdateListener;
+import com.google.firebase.remoteconfig.ConfigUpdateListenerRegistration;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ConfigRealtimeHttpClient {
+
+  private final Map<Integer, ConfigUpdateListener> listeners;
+  private int listenerCount;
+
+  public ConfigRealtimeHttpClient() {
+    listeners = new HashMap<>();
+    listenerCount = 1;
+  }
+
+  // Kicks off Http stream listening and autofetch
+  private void beginRealtime() {}
+
+  // Pauses Http stream listening
+  private void pauseRealtime() {}
+
+  public ConfigUpdateListenerRegistration addRealtimeConfigUpdateListener(
+      ConfigUpdateListener configUpdateListener) {
+    listeners.put(listenerCount, configUpdateListener);
+    beginRealtime();
+    return new ConfigUpdateListenerRegistrationInternal(this, listenerCount++);
+  }
+
+  public void removeRealtimeConfigUpdateListener(int listenerKey) {
```

Comment:
private?

Reply:
Left public so registration could access but fixed with above change. Changed to private.